infra/lib/auth/stack.create-user.ts

```diff
@@ -35,123 +35,125 @@ export const handler: lambda.AppSyncResolverHandler<
+      }
+      if (input.givenName) {
+        UserAttributes.push({ Name: "given_name", Value: input.givenName });
+      }
+      const createUserParams: cognito.AdminCreateUserCommandInput = {
         ...baseParams,
-        Password: input.password,
-        Permanent: true,
+        TemporaryPassword: input.password,
+        UserAttributes,
       };
       const resp = await CLIENT.send(
-        new cognito.AdminSetUserPasswordCommand(setPasswordParams)
-      );
-      console.log(
-        `Updated password for ${username} to ${input.password}`,
-        resp
+        new cognito.AdminCreateUserCommand(createUserParams)
       );
+      console.log(`Successfully created user: ${username}`, resp);
+      cognitoUsername = resp.User!.Username!;
     } catch (err: any) {
-      console.log(`Could not update password for ${username}`, err);
+      console.error(`Could not create user ${username}`, err);
       return {
         success: false,
         error: err.toString(),
       };
     }
-  }
 
-  if (input.enableMFA) {
-    console.log(`Enabling MFA for ${username}...`);
-    try {
-      const mfaParams: cognito.AdminSetUserMFAPreferenceCommandInput = {
-        ...baseParams,
-        SMSMfaSettings: {
-          Enabled: true,
-          PreferredMfa: true,
-        },
-      };
-      const resp = await CLIENT.send(
-        new cognito.AdminSetUserMFAPreferenceCommand(mfaParams)
-      );
-      console.log(`Successfully enabled MFA for ${username}`, resp);
-    } catch (err: any) {
-      console.log(`Could not enable MFA for ${username}`, err);
-      return {
-        success: false,
-        error: err.toString(),
-      };
+    if (autoConfirm) {
+      console.log(`Updating password for ${username}...`);
+      try {
+        const setPasswordParams: cognito.AdminSetUserPasswordCommandInput = {
+          ...baseParams,
+          Password: input.password,
+          Permanent: true,
+        };
+        const resp = await CLIENT.send(
+          new cognito.AdminSetUserPasswordCommand(setPasswordParams)
+        );
+        console.log(
+          `Updated password for ${username} to ${input.password}`,
+          resp
+        );
+      } catch (err: any) {
+        console.log(`Could not update password for ${username}`, err);
+        return {
+          success: false,
+          error: err.toString(),
+        };
+      }
     }
-  }
 
-  if (input.verifyAttributes) {
-    console.log(`Verifying attributes for ${username}...`);
-    try {
-      const userAttributesParams: cognito.AdminUpdateUserAttributesCommandInput =
+    if (enableMFA) {
+      console.log(`Enabling MFA for ${username}...`);
+      try {
+        const mfaParams: cognito.AdminSetUserMFAPreferenceCommandInput = {
+          ...baseParams,
+          SMSMfaSettings: {
+            Enabled: true,
+            PreferredMfa: true,
+          },
+        };
+        const resp = await CLIENT.adminSetUserMFAPreference(mfaParams);
```

Comment:
```suggestion
        const resp = await FULL_CLIENT.adminSetUserMFAPreference(mfaParams);
```
This is the only call that requires the new constructor.